### PR TITLE
yaml definition for deploying into a k8s cluster

### PR DESCRIPTION
**Pull request by juandebravo**
- define the `call-roulette` namespace
- define the `call-roulette` deployment. Only 1 replica (no HA) and
  using the docker container `juandebravo/call-roulette:0.2` (no https)
- ~create the `call-roulette` service, type `NodePort` (minikube does
  not support `LoadBalancer` type yet).~
- Include ingress controller and ingress rules

### CODE REVIEW

**Comment by albertobaselga** [comment]
What about including [ingres srules](https://kubernetes.io/docs/user-guide/ingress/) to manage traffic from outside?

File: deploy/k8s-cheat-sheet.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+*         minikube            minikube          minikube                call-roulette
+
+```
+
+### Deployment
+
+Deploy an app:
+
+```bash
+> kubectl run <deployment-name> --image=<docker-image>:<image-version> --port=<port-to-be-used-by-the-app>
```
Comment (albertobaselga) [inline deploy/k8s-cheat-sheet.md:77]: You can also use `kubectl create -f /path/to/call-roulette-deploy.yaml` to create the Deployment resource.

Comment (juandebravo) [reply deploy/k8s-cheat-sheet.md:77]: yes, it's in the README -> https://github.com/saghul/CallRoulette/pull/7/files#diff-bcdcc71d7dab57472393a413b9f8293dR32

Comment (dannyk81) [reply deploy/k8s-cheat-sheet.md:77]: better use `apply` instead of `create`, especially for Deployments :)

Comment (dannyk81) [reply deploy/k8s-cheat-sheet.md:77]: and indeed you do in the README, kudos!

File: deploy/call-roulette-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        kubernetes.io/cluster-service: "true"
+    spec:
+      terminationGracePeriodSeconds: 60
+      # volumes:
+      # - name: dhparam-example
+      #   secret:
+      #     secretName: dhparam-example
+      containers:
+        # Check 0.9.0-beta.3
+      - image: gcr.io/google_containers/nginx-ingress-controller:0.8.3
```
Comment (dannyk81) [inline deploy/call-roulette-deploy.yaml]: go for 0.9.0-beta.3 :)

File: deploy/call-roulette-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  labels:
+    k8s-app: nginx-ingress-lb
+    kubernetes.io/cluster-service: "true"
+spec:
+  replicas: 1
+  selector:
+    k8s-app: nginx-ingress-lb
+    kubernetes.io/cluster-service: "true"
+  template:
+    metadata:
```
Comment (dannyk81) [inline deploy/call-roulette-deploy.yaml]: To be "Prometheus Ready" :) add the following to the template metadata:
```
      annotations:
        prometheus.io/port: '10254'
        prometheus.io/scrape: 'true'
```
and you'll also need to include a configmap for the ingress controller with `enable-vts-status: true`

```
apiVersion: v1
data:
  enable-vts-status: "true"
kind: ConfigMap
metadata:
  name: nginx-ingress-lb-conf
  namespace: kube-system
```

make sure to load the custom config by adding the following to `args`:
`- --configmap=$(POD_NAMESPACE)/nginx-ingress-lb-conf`

File: deploy/call-roulette-deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        - containerPort: 80
+          hostPort: 80
+        - containerPort: 443
+          hostPort: 443
+        # volumeMounts:
+        # - mountPath: /etc/nginx-ssl/dhparam
+        #   name: dhparam-example
+        args:
+        - /nginx-ingress-controller
+        - --default-backend-service=call-roulette/call-roulette
```
Comment (dannyk81) [inline deploy/call-roulette-deploy.yaml]: load your custom configmap here:
`- --configmap=$(POD_NAMESPACE)/nginx-ingress-lb-conf`

**Pull request merged**